### ISSUE TITLE
vim syntax plugin?

### ISSUE THREAD

**Issue by benoitc** [inaka/erlang_guidelines#69]
do you know any vim syntax plugin that handle all the guidelines? or how to configure vim-erlang?

**Comment by elbrujohalcon** [comment]
I don't, but maybe @igaray (a vim user) knows?

**Comment by igaray** [comment]
I previously used vimerl (defunct and now superceded by https://github.com/vim-erlang) and Csaba's plugin (can't find it, maybe @elbrujohalcon can ask), but guideline enforcement was manual, except for setting indentation to 2 spaces for most projects.

**Comment by igaray** [comment]
@benoitc in any case, if it's of any interest to you at all, [here](https://github.com/igaray/dotfiles/blob/master/vim/.vimrc) is my vim config, but it's not erlang nor erlang-guideline specific.

**Comment by benoitc** [comment]
@igaray ok thanks. looking at it

**Comment by cloud8421** [comment]
@benoitc I'm using a few plugins to have syntax, some generators and inline syntax checks, see [here](https://github.com/fully-forged/dotfiles/blob/master/nvim/.config/nvim/init.vim#L39-L44) for their names (they're all well documented - @elbrujohalcon they're the ones from @hcs42 )

**Comment by hcs42** [comment]
[The vim-erlang-runtime][2] repository contains the syntax highlight and indentation plugins for Vim. These plugins have been included in Vim since 2013 and haven't changed much since late 2014, so you don't need to install them if you have a reasonably recent Vim.

The syntax highlight plugin doesn't have much to do with coding conventions. It tries its best to highlight any Erlang source file; it won't highlight possible problems or instances of breaking the guideline.

The indentation follows the Emacs indentation style ([with a few notable exceptions][1]), which is a convention in itself. The Inaka guidelines mention only two things regarding indentation ("Spaces over tabs, 2 space indentation"). These are both configurable on the Vim level, and any indentation plugin will respect that.

In short, there is no Vim plugin in vim-erlang related to these guidelines.

[1]: https://github.com/vim-erlang/vim-erlang-runtime/issues/31#issuecomment-189942651
[2]: https://github.com/vim-erlang/vim-erlang-runtime
